src/core/ext/xds/xds_client_stats.h

```diff
@@ -215,27 +216,32 @@ class XdsClusterLocalityStats : public RefCounted<XdsClusterLocalityStats> {
   Snapshot GetSnapshotAndReset();
 
   void AddCallStarted();
-  void AddCallFinished(bool fail = false);
+  void AddCallFinished(const std::map<absl::string_view, double>* named_metrics,
```

Comment:
> Suggest making this parameter a reference instead of a pointer.

I switched to a pointer to make it nullable so that I can avoid allocating an empty map when there's no load report, which I think is worth it here. From what I read GetBackendMetricData() wil return null if "endpoint-load-metrics-bin" trailing metadata is missing.

Reply:
But it requires allocating an empty string unnecessarily unless we make this nullable in some way.
In XdsClusterImplLb::Picker::SubchannelCallTracker::Finish() where we call AddCallFinished():

```
auto* backend_metric_data = args.backend_metric_accessor->GetBackendMetricData();
const std::map<absl::string_view, double>* named_metrics = nullptr;
if (backend_metric_data != nullptr) {
  named_metrics = &backend_metric_data->named_metrics;
}
locality_stats_->AddCallFinished(named_metrics, !args.status.ok());
```

vs. Something like (pseudo-code)
```
auto* backend_metric_data = args.backend_metric_accessor->GetBackendMetricData();
std::map<absl::string_view, double> named_metrics
if (backend_metric_data == nullptr) {
  locality_stats_->AddCallFinished(std::map<..>(), !args.status.ok());
} else {
  locality_stats_->AddCallFinished(
    backend_metric_data->named_metrics, !args.status.ok());
}
```

Reply:
I'll keep it then. Certainly no dynamic allocation, but I found this option looks slightly better: https://screenshot.googleplex.com/B5csTZpq6Eda2Gv

Reply:
Note that the first version in that screenshot is not the same as the version that I proposed in my last comment.  The version you're using imposes the overhead of a `std::move()`, which is not present in the version I proposed, so I think it should have less overhead.

In any case, though, as I said, this isn't worth holding up this PR over.

Reply:
Right that we should only compare the empty case here, and it comes down to https://screenshot.googleplex.com/329NtyXoDS7yPZX. That said, I overestimated the cost of allocating an empty map in the stack which should be very cheap, however like you said I think we can keep it, which won't hurt. I will resolve thisl.